providers/gce/gce_loadbalancer_internal.go

```diff
@@ -254,6 +256,37 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 	return status, nil
 }
 
+func filterOutNodesFromNonDefaultNetworks(nodes []*v1.Node, defaultSubnetURL string) []*v1.Node {
+	var newList []*v1.Node
+	for _, node := range nodes {
+		subnetLabel, ok := node.Labels[labelGKESubnetworkName]
+
+		if !ok && node.Spec.PodCIDR == "" {
+			continue
+		}
+		if ok && subnetLabel != "" && subnetLabel != subnetNameFromURL(defaultSubnetURL) {
+			continue
+		}
+
+		newList = append(newList, node)
+	}
+	return newList
+}
+
+// Extract the subnet name from the URL.
+// example: for `https://www.googleapis.com/compute/v1/projects/project/regions/us-central1/subnetworks/defaultSubnet`
+// this will return `defaultSubnet`.
+func subnetNameFromURL(url string) string {
```

Comment:
This should have its own unit test.

This func should return an error instead of empty string if it fails to parse instead of eating the error.

Reply:
changed it